go/oasis-test-runner/scenario/e2e/runtime/keymanager_key_generation.go

```diff
@@ -0,0 +1,236 @@
+package runtime
+
+import (
+	"context"
+	"fmt"
+
+	beacon "github.com/oasisprotocol/oasis-core/go/beacon/api"
+	"github.com/oasisprotocol/oasis-core/go/common"
+	"github.com/oasisprotocol/oasis-core/go/common/cbor"
+	consensus "github.com/oasisprotocol/oasis-core/go/consensus/api"
+	"github.com/oasisprotocol/oasis-core/go/oasis-test-runner/env"
+	"github.com/oasisprotocol/oasis-core/go/oasis-test-runner/oasis"
+	"github.com/oasisprotocol/oasis-core/go/oasis-test-runner/scenario"
+)
+
+// KeymanagerKeyGeneration is the keymanager key generation scenario.
+//
+// It uses encryption and decryption transactions provided by the
+// simple key/value runtime to test whether the key manager client
+// can retrieve private and public ephemeral keys from the key manager
+// and if the latter generates those according to the specifications.
+var KeymanagerKeyGeneration scenario.Scenario = newKmKeyGenerationImpl()
+
+type kmKeyGenerationImpl struct {
+	runtimeImpl
+}
+
+func newKmKeyGenerationImpl() scenario.Scenario {
+	return &kmKeyGenerationImpl{
+		runtimeImpl: *newRuntimeImpl("keymanager-key-generation", BasicKVEncTestClient),
+	}
+}
+
+func (sc *kmKeyGenerationImpl) Fixture() (*oasis.NetworkFixture, error) {
+	return sc.runtimeImpl.Fixture()
+}
+
+func (sc *kmKeyGenerationImpl) Clone() scenario.Scenario {
+	return &kmKeyGenerationImpl{
+		runtimeImpl: *sc.runtimeImpl.Clone().(*runtimeImpl),
+	}
+}
+
+func (sc *kmKeyGenerationImpl) Run(childEnv *env.Env) error {
+	ctx := context.Background()
+	if err := sc.runtimeImpl.startNetworkAndTestClient(ctx, childEnv); err != nil {
+		return err
+	}
+
+	// Wait for the client to exit.
+	if err := sc.waitTestClientOnly(); err != nil {
+		return err
+	}
```

Comment:
Can remove the test client for this test case. Use just `sc.startNetworkAndWaitForClientSync` instead.

Reply:
Thought it was a good practice to always run the client just to be sure that everything works. But I guess we don't need this in this case as we are already testing it.